enhancements/operator-hub-check.md

```diff
@@ -0,0 +1,103 @@
+---
+title: operatorhub-bundle-validation
+authors:
+  - "@camilamacedo86"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-19
+last-updated: 2021-02-19
+status: implementable
+---
+
+# Add checks to OperatorHubValidator
+
+This proposal describes a new checks to be implemented in the [operator-framework/api][oper-api] into the [OperatorHubValidator][operatorhubvalidator].
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+`OperatorHubValidator` checks if the operator is respecting the required criteria to be published in the [OperatorHub.io](https://operatorhub.io/). This proposal describes criteria to be added to it.
+
+Many tools and projects use `OperatorHubValidator` already. Note that the new checks described by this proposal will be available via SDK:
+
+```sh
+To validate a bundle with a superset of requirements for operatorhub.io specifically:
+   
+  $ operator-sdk bundle validate ./bundle --select-optional name=operatorhub
+``` 
+
+**NOTE** For further information check [here](https://sdk.operatorframework.io/docs/olm-integration/generation/#validation) in the SDK docs and the [here](https://olm.operatorframework.io/docs/tasks/validate-package/#validation) in the OLM docs. Also, the new checks are demanding to address a further requirement proposal to audit all bundles of a provided catalogue. 
+
+## Motivation
+
+- Allow operator developers and automatically jobs check required criteria to publish an operator bundle on OperatorHub. 
+- Validate the operator bundle manifests and notify users and maintainers with warning and errors about what is not compliant to the OperatorHub criteria
+
+### Goals
+
+- Perform static checks on the operator bundle manifests to avoid the scenarios where the operator bundle:
+    - is published but will not be installable
+    - is not prepared to work well on all Kubernetes cluster versions and has not any configuration to address its limitation
+    - is using deprecated or no longer supported K8S APIs
+    - is not following the good practices and conventions defined 
+- [operator-framework/api][oper-api] with new checks in `OperatorHubValidator`
+- SDK tool using [operator-framework/api][oper-api] and return errors or warnings regarding the new checks via `operator-sdk bundle validate ./bundle --select-optional name=operatorhub`
```

Comment:
We should add more to this enhancement delineating what will be warned and what will be errored on. The distinction will be critical if this is called in the CI pipeline, as WARN level will likely just be something you look at in the gathered log and event output of a job, whereas ERROR level output will likely mean a failed job. Either level could be appropriate, it's just that we should give some mention here as to what types of things would likely be what level.

Reply:
I tried to address your concern here. Please, let me know wdyt